### PR TITLE
Don't rely on try/catch only for prior crash fix (performance)

### PR DESCRIPTION
**Pull request by HermanSchoenfeld**

### CODE REVIEW

File: NBitcoin/Protocol/Node.cs [hunk 1]
```diff
@@ -161,12 +161,13 @@ public void BeginListen()
...
-						}
-						catch { }
-					};
+                        try {
+                            if (!ar.SafeWaitHandle.IsClosed && !ar.SafeWaitHandle.IsInvalid)
+                                ar.Set();
+                        } catch (Exception error) {
+                            // no throw
+                        }
+                    };
```
Comment (knocte) [inline NBitcoin/Protocol/Node.cs]: You're using spaces here, and the previous code was using tabs.

File: NBitcoin/Protocol/Node.cs [hunk 2]
```diff
@@ -659,12 +660,13 @@ internal Node(NetworkAddress peer, Network network, NodeConnectionParameters par
...
-						}
-						catch { }
-					};
+					    try {
+					        if (!completed.SafeWaitHandle.IsClosed && !completed.SafeWaitHandle.IsInvalid)
+					            completed.Set();
+					    } catch {
+					        // no throw
+					    }
+                    };
```
Comment (knocte) [inline NBitcoin/Protocol/Node.cs]: This line is using spaces while the other ones use tabs.

**Comment by knocte** [comment]
Cool, but it's better to not have commits with just indentation changes, so can you squash your commits into one? Easy way is `git rebase --interactive HEAD~2` and then mark the 'convert to tabs' commit as "squash" instead of pick -> done

**Comment by HermanSchoenfeld** [comment]
I followed your rebase instructions but I'm not sure if that reduced the commits. I'm not that familiar with rebase unfortunately.

**Comment by NicolasDorier** [comment]
```
git rebase -i HEAD~5
```
* Type i for going into inset mode,
* Replace "pick" by "f" in from of commit you want to "merge"
* ESCAPE, then type !wq

Push with "git push --force"

**Comment by HermanSchoenfeld** [comment]
Can you use Remote Viewer now into my machine and do it please..

ID: 856 438 216
Password: 852uci


Regards,
Herman Schoenfeld	M  +61 437 553 144
AU	T  +61 7 3195 3224	F  +61 7 3188 6009
US	T  +1 (562) 661 8263	F  +1 (562) 661 8263
2/2902 Logan Rd, Underwood QLD 4119


 <http://www.sphere10.com/>
 <http://www.sphere10.com/>

**Comment by HermanSchoenfeld** [comment]
ID: 856 438 216
Password: 45i1mp

Regards,
Herman Schoenfeld	M  +61 437 553 144
AU	T  +61 7 3195 3224	F  +61 7 3188 6009
US	T  +1 (562) 661 8263	F  +1 (562) 661 8263
2/2902 Logan Rd, Underwood QLD 4119


 <http://www.sphere10.com/>
 <http://www.sphere10.com/>

**Comment by NicolasDorier** [comment]
a sec downloading it

**Comment by NicolasDorier** [comment]
Don't forget to change the password/id now :p

**Comment by HermanSchoenfeld** [comment]
Thanks again for remote tut (password changes every session).

**Pull request merged**